recomputeExpression.php

```diff
@@ -69,17 +70,25 @@ public function afterPluginLoad()
         if (Yii::app() instanceof CConsoleApplication) {
             return;
         }
-        if(strpos(Yii::app()->request->getRequestUri(),'admin/responses')){// Hack to register only on browse response
-            $surveyid = Yii::app()->request->getparam('surveyid');
-            $responseid = Yii::app()->request->getparam('id');
-            $aRecomputeVar=array(
-                'jsonurl'=>Yii::app()->getController()->createUrl('plugins/direct', array('plugin' => 'recomputeExpression', 'function' => 'recompute')),
-                'surveyId'=>$surveyid,
-                'responseId'=>$responseid,
-            );
-            Yii::app()->getClientScript()->registerScript('aRecomputeVar','recomputeVar='.json_encode($aRecomputeVar),CClientScript::POS_BEGIN);
-            Yii::app()->getClientScript()->registerScriptFile(Yii::app()->assetManager->publish(dirname(__FILE__) . '/js/recompute.js'));
-        }
+        // Hack to register only on browse response
+        $event      = $this->getEvent();
+
+        //$surveyid = Yii::app()->request->getparam('surveyId');
+        $surveyid = Yii::app()->request->getparam('surveyId');
+
+        //$responseid = Yii::app()->request->getparam('id');
+        $responseid = Yii::app()->request->getparam('id');
+
+        $aRecomputeVar=array(
+            'jsonurl'=>Yii::app()->getController()->createUrl('plugins/direct', array('plugin' => 'recomputeExpression', 'function' => 'recompute')),
+            'surveyId'=> $surveyid,
+            'responseId' => $responseid,
+            'isResponsePage' => true,
+            'route' => Yii::app()->request->getPathInfo()
```

Comment:
Unsure on this point : [can be /limesurvey](https://www.yiiframework.com/doc/api/1.1/CHttpRequest#getPathInfo-detail)

Maybe better to use (new event) : https://manual.limesurvey.org/BeforeControllerAction and register at this moment (didn't exist)